analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -211,22 +227,56 @@ any(duplicated(rna_df$sample_id))
 ```
 What are they?
 
+Checking for duplicate rna sample_ids
+
 ```{r}
 histologies_df %>%
   filter(sample_id %in% rna_df$sample_id[which(duplicated(rna_df$sample_id))],
          experimental_strategy == "RNA-Seq") %>%
   group_by(sample_id) %>%
-  summarize(RNA_library = paste(RNA_library, collapse = ", "))
+  summarize(RNA_library = paste(RNA_library, collapse = ", "),
+            composition =  paste(composition, collapse = ", "))
 ```
 
-There are 2 instances where there are two stranded biospecimen IDs that map to the same `sample_id`.
+The duplicates are mostly from multiple Solid tumor samples per sample_id or there are non-tumor ( derived cell-line ) samples for sample_id 
+
+
+Checking if adding cell-line composition type can uniquely match DNA to RNA
+
+```{r}
+
+cell_line_composition <- read_tsv("input/cell-line-composition.tsv",col_types = 
+                                    readr::cols( aliquot_id  = readr::col_character()))
+
+cell_line_composition %>%
+  left_join(histologies_df) %>%
+  group_by(sample_id,experimental_strategy) %>%
+  summarize(cell_line_composition = paste(cell_line_composition, collapse = ", "),
+            composition =  paste(composition, collapse = ", "))
+
+```
+`cell_line_composition` seems to be able to differentiate each cell line so let's add the column to merged dna and rna files
+
+```{r}
+
+dna_df <- dna_df %>%
+  left_join(cell_line_composition)
+
+rna_df <- rna_df %>%
+  left_join(cell_line_composition)
```

Comment:
Can you explicitly add  `by = c("Kids_First_Participant_ID", "sample_id", "Kids_First_Biospecimen_ID")` to these please?